src/ImageSharp/Formats/Gif/GifEncoderCore.cs

```diff
@@ -326,15 +324,24 @@ private int GetTransparentIndex<TPixel>(IndexedImageFrame<TPixel> quantized)
         /// Writes the application extension to the stream.
         /// </summary>
         /// <param name="stream">The stream to write to.</param>
+        /// <param name="frameCount">The frame count fo this image.</param>
         /// <param name="repeatCount">The animated image repeat count.</param>
-        private void WriteApplicationExtension(Stream stream, ushort repeatCount)
+        /// <param name="xmpProfile">The XMP metadata profile. Null if profile is not to be written.</param>
+        private void WriteApplicationExtensions(Stream stream, int frameCount, ushort repeatCount, XmpProfile xmpProfile)
         {
-            // Application Extension Header
+            // Application Extension: Loop repeat count.
             if (repeatCount != 1)
```

Comment:
Looks like you're not actually using the frameCount. 

```c#
 if (frameCount > 1 && repeatCount != 1)
```

Reply:
Oops. Checking it now.